diffprivlib/models/forest.py

```diff
@@ -612,6 +617,39 @@ def apply(self, X):
 
         return out
 
+class DifferentiallyPrivateRandomForestRegressor(RandomForestRegressor):
+    def __init__(self, n_estimators=100, epsilon=1.0, bounds=None, random_state=None, **kwargs):
+        super().__init__(n_estimators=n_estimators, **kwargs)
+        self.epsilon = epsilon
+        self.bounds = bounds
+        self.random_state = random_state
+
+    def _dp_mean(self, array):
+        epsilon = self.epsilon  # Privacy parameter
+        bounds = self.bounds if self.bounds else (np.min(array), np.max(array))  # Bounds of the array
+        
+        # Compute sensitivity (max possible change in output when a single record is added or removed)
+        sensitivity = np.abs(bounds[1] - bounds[0]) / len(array)
+
+        # Create Laplace noise mechanism
+        laplace_mechanism = Laplace(epsilon=epsilon, sensitivity=sensitivity, random_state=self.random_state)
+
+        # Compute differentially private mean
+        private_mean = np.mean(array) + laplace_mechanism.sample()
+        return private_mean
+
+    def fit(self, X, y):
+        super().fit(X, y)  # Fit the RandomForestRegressor
```

Comment:
Here you are fitting with the parent RandomForrestRegressor which means no differential privacy will be applied during the training (ie the main point). Please go back an look at how they created the RandomForrestClassifier via custom trees. Their trees randomly partition the domain (not using the gini or entropy coefficient like in the vanilla RandomForrestRegressor). Then once the random trees are created you can use your _dp_mean method on the leaves.

Reply:
please check now